app/models/notification_blueprint.rb

```diff
@@ -0,0 +1,23 @@
+# Model to store all information related to a notification.
+# The audience for this notification is calculated on real time by
+# Notification.
+#
+# Foreman and plugins should define NotificationBlueprints for various
+# actions that are notification-worthy. This keeps the responsibilities
+# separate, Notifications taking care of serving the content, and
+# NotificationBlueprint of storing it.
+class NotificationBlueprint < ActiveRecord::Base
+  has_many :notifications
+  belongs_to :subject, :polymorphic => true
+
+  validates :message, :presence => true
+  validates :level, :inclusion => { :in => %w(success error warning info) }, :presence => true
+  validates :expires_in, :numericality => {:greater_than => 0}
```

Comment:
I forgot to validate `group`, it seems that the patternfly UI doesn't allow notifications without group. I'll add it.

For `name` I can do so too, but if it's a field that's not going to be displayed in the UI and it isn't really used for anything that dev. convenience, why should we enforce having it?